### PR TITLE
Job issue 164

### PR DESCRIPTION
**Pull request by mohitgusain**
- added a new data field due_date.
- due date field available in invoice creation page.

Commands to run:

```
> php artisan migrate
```

### CODE REVIEW

**Comment by rathorevaibhav** [comment]
@mohitgusain this PR has conflicts. Please resolve.

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -13,7 +14,7 @@ class AppServiceProvider extends ServiceProvider
      */
     public function boot()
     {
-        //
+     Schema::defaultStringLength(191);   //
```
Comment (rathorevaibhav) [inline app/Providers/AppServiceProvider.php]: please correct the indentation. Laravel follows PSR2 guidelines for code formatting.

Read more on PSR2 here: https://www.php-fig.org/psr/psr-2/

File: database/migrations/2018_05_23_165429_update_hr_rounds_add_round_mail_templates.php [hunk 2]
```diff
@@ -14,8 +14,8 @@ class UpdateHrRoundsAddRoundMailTemplates extends Migration
     public function up()
     {
         Schema::table('hr_rounds', function(Blueprint $table) {
-            $table->json('confirmed_mail_template')->nullable()->after('guidelines');
-            $table->json('rejected_mail_template')->nullable()->after('confirmed_mail_template');
+            $table->text('confirmed_mail_template')->nullable()->after('guidelines');
```
Comment (rathorevaibhav) [inline database/migrations/2018_05_23_165429_update_hr_rounds_add_round_mail_templates.php]: I don't think we want this change. We're utilizing MySQL's json support for which Laravel5.6 also provides additional support to code. If you're having a lower version of MySQL, please update to MySQL 5.7

File: database/migrations/2018_06_04_132230_due_date.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+       Schema::table('finance_invoices', function (Blueprint $table) {
+        $table->date('due_date');
```
Comment (rathorevaibhav) [inline database/migrations/2018_06_04_132230_due_date.php]: What would be the default date? Clearly, this date will not be present for existing invoices. It will give an error.

Comment (rathorevaibhav) [reply database/migrations/2018_06_04_132230_due_date.php]: Also, correct the indentation.

File: database/migrations/2018_06_04_132230_due_date.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        //
```
Comment (rathorevaibhav) [inline database/migrations/2018_06_04_132230_due_date.php]: You also need to write the code to revert these changes. This'll help if some errors happen during deployment and we want to revert the changes. No code here, will not undo the changes this migration did.

File: resources/views/finance/invoice/create.blade.php [hunk 5]
```diff
@@ -203,6 +203,10 @@
                     <div class="form-group col-md-2" v-show="paymentType == 'cheque' && chequeStatus == 'cleared'">
                         <label for="cheque_cleared_date">Cheque Cleared Date</label>
                         <input type="text" class="form-control date-field" name="cheque_cleared_date" id="cheque_cleared_date" placeholder="dd/mm/yyyy" value="{{ old('cheque_cleared_date') ? date(config('constants.display_date_format'), strtotime(old('cheque_cleared_date'))) : '' }}">
+                    </div>
+                     <div class="form-group col-md-2" v-show="paymentType == 'cheque' && chequeStatus == 'received'">
```
Comment (rathorevaibhav) [inline resources/views/finance/invoice/create.blade.php]: I think there is a confusion in your understanding of the requirement. This field is not related to cheque. This is a due date for invoice. This field represents the date after which, if the invoice isn't paid, we need to remind the client. 
It should be another form field like `sent_on` date.

File: resources/views/hr/job/edit.blade.php [hunk 6]
```diff
@@ -67,6 +67,7 @@ class="form-control"
 
             <div class="card-footer">
                 <button type="submit" class="btn btn-primary">Update</button>
+                <a href="{{ $job->link }}" target="_blank" class="btn btn-info" role="button">Preview job</a>
```
Comment (rathorevaibhav) [inline resources/views/hr/job/edit.blade.php]: Is this change relevant to this issue? if not, please track it in a separate branch.

File: resources/views/hr/job/index.blade.php [hunk 7]
```diff
@@ -21,9 +20,7 @@
             <td>
                 {{ ucfirst($job->type) }}
             </td>
-            <td>
```
Comment (rathorevaibhav) [inline resources/views/hr/job/index.blade.php]: do these changes required to fix this issue?

**Comment by mohitgusain** [comment]
Please check my corrections on this issue @rathorevaibhav

File: database/migrations/2018_06_04_132230_due_date.php (hunk 3 above)
Comment (rathorevaibhav) [reply database/migrations/2018_06_04_132230_due_date.php]: @mohitgusain line 17 is not properly indented. Please correct.

File: database/migrations/2018_06_04_132230_due_date.php (hunk 4 above)
Comment (rathorevaibhav) [reply database/migrations/2018_06_04_132230_due_date.php]: @mohitgusain I don't see any comment/update on this feedback. Do you have any doubts?

File: database/migrations/2018_06_04_132230_due_date.php (hunk 3 above)
Comment (mohitgusain) [reply database/migrations/2018_06_04_132230_due_date.php]: sir i did not add this file for correction and i removed it because it shows some errors when i was  creating new migration. So for this i create a new migration of same name of  due_date  and  the name is 2018_06_08_093401_due_date.php and i  push it for review @rathorevaibhav

**Comment by bohraji** [comment]
@mohitgusain there are now two migrations doing the same thing right now. Please check and remove the redundant one.

**Comment by mohitgusain** [comment]
redundant migration file is removed @bohraji , @rathorevaibhav
